timestream.lua

```diff
@@ -0,0 +1,339 @@
+-- speeds up the calendar, units, or both
+
+--[====[
+
+timestream
+==========
+This is only for fortress mode.
+
+Usage:
+:timestream <scalar> <fps> <simulate units y/n>
+Examples:
+
+:timestream     2:          Calendar runs at x2 normal speed
+:timestream    -1 100:      Calendar runs at dynamic speed to simulate 100 FPS
+:timestream    -1 100 1:    Calendar & units are simulated at 100 FPS
+
+]====]
+
+args={...}
+local rate=tonumber(args[1])
+local desired_fps = tonumber(args[2])
+local simulating_units = tonumber(args[3])  -- Setting this to 2 instead of 1 will use debug_turbospeed instead of adjusting the timers of all creatures. I don't quite like it because it makes everyone move like insects.
+local debug_mode = tonumber(args[4])
+
+local minimal_fps = 10 -- This ensures you won't get crazy values on pausing/saving, or other artefacts on extremely low FPS.
+local splicing_threshold = 0.25 -- This controls how generously the script decides to speed up units. Higher values mean that units will be sped up more often than not. This is only for when using debug_turbospeed.
+
+local current_fps = desired_fps
+local prev_tick = 0
+local ticks_left = 0
+local simulating_desired_fps = false
+local prev_frames = df.global.world.frame_counter
+local last_frame = df.global.world.frame_counter
+local prev_time = df.global.enabler.clock
+
+if rate == nil then
+    rate = 1
+elseif rate < 0 then
+    rate = 0
+end
+
+if desired_fps == nil or desired_fps <= 0 then
+    desired_fps = 100
+    current_fps = desired_fps
+end
+
+if debug_mode == nil or debug_mode ~= 1 then
+    debug_mode = false
+else
+    debug_mode = true
+end
+
+eventNow = false
+seasonNow = false
+timestream = 0
+counter = 0
+if df.global.cur_season_tick < 3360 then
```

Comment:
I realize you didn't write some of this, but it would be nice to use constants instead of magic numbers. I think I see how most of these correspond to units from https://dwarffortresswiki.org/index.php/DF2014:Time (I'm guessing `cur_season_tick` is `cur_year_tick / 10`?) but it would be easier to read through if they had names instead.

Reply:
Done

Reply:
That looks good, thanks!